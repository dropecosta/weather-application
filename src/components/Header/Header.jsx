import React, { useState } from 'react';
import ThemeSwitch from '../ThemeSwitch/ThemeSwitch';
import sun from '../../assets/icons/sun.svg';
import moon from '../../assets/icons/moon.svg';
import './Header.scss';

const Header = () => {
  const [darkMode, setDarkMode] = React.useState(false);
  const [checked, setChecked] = useState(false);

  React.useEffect(() => {
    const body = document.body;
    const toggle = document.querySelector('.toggle-inner');

    if (darkMode === true) {
      body.classList.add('dark-mode');
    toggle?.classList.add('toggle-active');
    } else {
      body.classList.remove('dark-mode');
      toggle?.classList.remove('toggle-active');
    }
  }, [darkMode]);

  return (
    <div className="header">
      <h2>Weather APP</h2>

      <div className="toggle-container">
      <img src={moon} className='darkmode-icon-moon' alt="moon"/>
        
        <ThemeSwitch
          isCheck={checked}
          onColor="#666"
          handleDarkMode={() => {
            setChecked(!checked);
            darkMode === false ? setDarkMode(true) : setDarkMode(false)
          }}
        />
        <img src={sun} className='darkmode-icon-sun' alt="sun"/>
      </div>
    </div>
  );
};

export default Header
