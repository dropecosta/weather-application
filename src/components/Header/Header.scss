.header {
    width: 100%;
    padding: 0 20px;
    display: flex;
    justify-content: space-between;
    
    h2 {
      font-size: 18px;
      font-weight: bold;
      text-transform: uppercase;
      margin-top: 3px;
    }
}

.label-description {
    text-transform: uppercase;
    font-size: 12px;
    padding: 2px 5px 0 0;
    
}

.toggle-group {
  width: 180px;
  height: 30px;
  display: flex;
  flex-direction: row-reverse;
}

.toggle {
    width: 50px;
    display: flex;
    padding: 1px 0 2px 2px;
    background-color: #1a202c;
    display: block;
    border-radius: 1000px;
    cursor: pointer;
    box-shadow: 0px 5px 20px -10px #000000;
    transition: background-color 0.2s ease-in;
    border: 1px solid #333;
    
    .toggle-inner {
      width: 15px;
      height: 15px;
      background-color: white;
      border-radius: 1000px;
      transition: margin-left 0.2s ease-in, background-color 0.2s ease-in;
  
      &.toggle-active {
        margin-left: 30px;
      }
    }
  }