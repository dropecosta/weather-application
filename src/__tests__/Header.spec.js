import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Header from "../components/Header/Header";

describe("Header component tests", () => {
  it("should render APP component with container class", () => {
    const { container } = render(<Header />);
    expect(container.firstChild).toHaveClass("header");
  });

  it("should match the snapshot", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render text content", () => {
    const { getByText } = render(<Header />);
    expect(getByText("Weather APP")).toBeInTheDocument();
  });
});
