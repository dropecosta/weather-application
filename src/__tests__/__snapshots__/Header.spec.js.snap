// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests should match the snapshot 1`] = `
<div
  className="header"
>
  <h2>
    Weather APP
  </h2>
  <div
    className="toggle-container"
  >
    <img
      alt="moon"
      className="darkmode-icon-moon"
      src="moon.svg"
    />
    <input
      checked={false}
      className="switch-checkbox"
      id="switch-darkmode"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="switch-label"
      htmlFor="switch-darkmode"
      style={
        Object {
          "background": false,
        }
      }
    >
      <span
        className="switch-button"
      />
    </label>
    <img
      alt="sun"
      className="darkmode-icon-sun"
      src="sun.svg"
    />
  </div>
</div>
`;
